components/accept-financial-contributions/HostsContainer.js

```diff
@@ -0,0 +1,137 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { Flex } from '@rebass/grid';
+import styled from 'styled-components';
+import { graphql } from '@apollo/react-hoc';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import HostCollectiveCard from './HostCollectiveCard';
+import Loading from '../Loading';
+import HorizontalScroller from '../HorizontalScroller';
+
+import withViewport, { isDesktopOrAbove } from '../../lib/withViewport';
+
+const limit = 12; // nice round number to make even rows of 2, 3, 4
+
+const AllCardsContainer = styled(Flex).attrs({
+  flexWrap: 'wrap',
+  width: '70%',
+  justifyContent: 'space-evenly',
+})``;
+
+const AllCardsContainerMobile = styled(Flex)`
+  overflow-x: auto;
+  scroll-behavior: smooth;
+  width: 100%;
+  padding: 16px;
+`;
+
+const CollectiveCardContainer = styled.div`
+  width: 280px;
+  padding: 20px 15px;
+`;
+
+class HostsContainer extends React.Component {
+  static propTypes = {
+    collective: PropTypes.object,
+    onChange: PropTypes.func,
+    data: PropTypes.object.isRequired,
+    viewport: PropTypes.string,
+    tags: PropTypes.string,
+  };
+
+  render() {
+    const { onChange, data, viewport } = this.props;
+
+    if (!data.hosts || !data.hosts.nodes) {
+      return <Loading />;
+    }
+
+    const hosts = [...data.hosts.nodes];
+
+    return (
+      <Fragment>
+        {!isDesktopOrAbove(viewport) ? (
```

Comment:
**[:sparkles: **Enhancement**]** As recently documented in https://github.com/opencollective/opencollective-frontend/pull/3959, using `withViewport` to handle responsiveness is discouraged if a CSS alternative is possible. The layout doesn't seems very different between mobile and desktop, the main difference is that in the first case they stay on the same line while on desktop they wrap into multiple lines. Sounds like something we could achieve with something like `flexWrap={['nowrap', null, 'wrap']}`?

---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>

Reply:
Just using 'nowrap' doesn't make it scroll horizontally though, which is why I was using the `HorizontalScroller` element. Maybe I can use that with `Hide` to achieve what I was trying to do with `withViewport`.

Reply:
Solved with Hide

Reply:
It's better than `withViewport` but keep in mind that it's rendering and updating both components (they're just hidden), which has a performance cost. Not a blocker, but we should ideally try to solves these kind of issues with flex (I don't know if it's possible in this specific case, I haven't digged).